app/controllers/providers/link_case/confirmations_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Providers
+  module LinkCase
+    class ConfirmationsController < ProviderBaseController
+      prefix_step_with :link_case
+
+      before_action :set_link_type_code,
+                    :set_linked_application_types,
+                    only: :show
+
+      def show
+        if legal_aid_application.copy_case?
+          destroy_linked_application
+          @form = ::LinkCase::ConfirmationForm.new(model: copied_application)
+        else
+          @form = ::LinkCase::ConfirmationForm.new(model: linked_application)
+        end
+      end
+
+      def update
+        @form = ::LinkCase::ConfirmationForm.new(form_params)
+
+        if @form.link_type_code.eql?("false")
+          destroy_linked_application
+        end
```

Comment:
I think this logic should be owned by the form and perform as part of save (after_validation perhaps, see comment below)